core/src/parser/expression.rs

```diff
@@ -50,32 +50,33 @@ pub fn ParseBinaryExpression(parser:Parser, previous: Expr) -> Expr {
 //     }
 // }
 
-pub fn ParsePrimaryExpression(parser:Parser) -> UnaryExpr {
-    let current = parser.peek().unwrap();
+pub fn ParsePrimaryExpression(parser:&mut Parser) -> UnaryExpr {
+    let Some(current) = parser.peek();
 
     //TODO: ParseConditional, ParseSelector
     let primary:PrimaryExpr = match current {
-        Token::BoolLit(current) => ParseBooleanLiteral(parser),
-        Token::IntLit(current) => ParseIntegerLiteral(parser),
-        Token::FloatLit(current) => ParseFloatLiteral(parser),
-        Token::CharLit(current) => ParseCharLiteral(parser),
-        Token::StrLit(current) => ParseStringLiteral(parser),
-        Token::Ident(current) => 
-            if parser.is_tok_curr(Token::Punc(Punc::LeftParen), parser.peek_offset(1)) {
+        Token::BoolLit(_current) => ParseBooleanLiteral(parser),
+        Token::IntLit(_current) => ParseIntegerLiteral(parser),
+        Token::FloatLit(_current) => ParseFloatLiteral(parser),
+        Token::CharLit(_current) => ParseCharLiteral(parser),
+        Token::StrLit(_current) => ParseStringLiteral(parser),
+        Token::Ident(_current) => 
+            if parser.is_tok_eq(Token::Punc(Punc::LeftParen), parser.peek_offset(1)) {
                 ParseFunctionCall(parser)
```

Comment:
There is no such thing as a function call according to the spec, only argument expressions